packages/bpk-component-nudger/src/BpkConfigurableNudger.js

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable react/forbid-prop-types */
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow strict */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import BpkButton from 'bpk-component-button';
+import { withButtonAlignment } from 'bpk-component-icon';
+import MinusIcon from 'bpk-component-icon/sm/minus';
+import PlusIcon from 'bpk-component-icon/sm/plus';
+import { cssModules } from 'bpk-react-utils';
+
+import {
+  type CommonProps,
+  COMMON_PROP_TYPES,
+  COMMON_DEFAULT_PROPS,
+} from './common-types';
+import STYLES from './BpkNudger.scss';
+
+const getClassName = cssModules(STYLES);
+
+const AlignedMinusIcon = withButtonAlignment(MinusIcon);
+const AlignedPlusIcon = withButtonAlignment(PlusIcon);
+
+type Props = {
```

Comment:
We should make this generic over `T` instead this way we can introduce more type safety than `any`. For example currently the following is valid

```js
const badIncrement = (current: number): string => {
  return "hello";
}
```

Here's a full [example](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgDieAdngE4CWAxgArlw4DOAPACoB8YAvGAN6oAkJRLVyeALakMANQCGMAK54AXGAAUbAJQ8ubADSoAvgG5MuAvUZMe-ISLGTp8pao1ySWHdy4esh03RqOBImDDAAIzkAEwBJUXEpEnDedWpFcnFktRJFCQiKLTUwqhIAc107QXEMDJIwACIACzwYeAazQNRg0PCy0goaK2Y1YjIqOgZmFlz8ii5eATAwQWBgMAASADF4BC3KAA8AWTx7BKdkl2U1KLjzpIwAs26QsLAcKaY1YZtFs8cHlc3Ld4gDpE9UEA)

Reply:
Ah this might be the thing me and shaun were looking for earlier! As the BpkNudger was throwing errors when passing the function through to here which is why we used any as the Generics didn't work

Reply:
This should definitely work

Reply:
@k0nserv we ran into issues and got stuck. I'll send you the code and see if you have any ideas

Reply:
We resolved this right?

Reply:
We did :)